File: livebox/src/main/java/com/creations/livebox/datasources/fetcher/Fetcher.java
package com.creations.livebox.datasources.fetcher;

import io.reactivex.Observable;

/**
 * @author Sérgio Serra on 25/08/2018.
 * Criations
 * sergioserra99@gmail.com
 */
public interface Fetcher<T> {
    Observable<T> fetch();
}

File: livebox/src/main/java/com/creations/livebox/datasources/fetcher/Fetcher.kt
package com.creations.livebox.datasources.fetcher

import io.reactivex.Observable
import io.reactivex.Single

/**
 * @author Sérgio Serra on 25/08/2018.
 * sergioserra99@gmail.com
 */
interface Fetcher<T> {
    fun fetch(): Observable<T>
}